rest/utilities_testing_resttester.go

```diff
@@ -491,3 +496,45 @@ func (rt *RestTester) PutDocWithAttachment(docID string, body string, attachment
 	}
 	return rt.PutDocDirectly(docID, rawBody)
 }
+
+type RawDocResponse struct {
+	Xattrs RawDocXattrs `json:"_xattrs"`
+}
+
+// RawDocXattrs is a non-exhaustive set of xattrs returned by the _raw endpoint. Used for test assertions.
+type RawDocXattrs struct {
+	RawDocXattrsWellKnown
+	RawDocXattrsOthers
+}
+
+// RawDocXattrsWellKnown contains well known fields - see RawDocXattrsOthers for those not known until runtime.
+type RawDocXattrsWellKnown struct {
+	Sync       db.SyncDataJSON `json:"_sync"`
+	GlobalSync map[string]any  `json:"_globalSync"`
+	VV         map[string]any  `json:"_vv"`
+	MOU        map[string]any  `json:"_mou"`
+}
+
+// RawDocXattrsOthers contains fields that can only be known at runtime (e.g. user xattr names)
+type RawDocXattrsOthers map[string]interface{}
+
+func (t *RawDocXattrs) MarshalJSON() ([]byte, error) {
+	// overwrite fields in Main into Extra
+	data, err := json.Marshal(t.RawDocXattrsWellKnown)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(data, &t.RawDocXattrsOthers); err != nil {
+		return nil, err
+	}
+	return json.Marshal(t.RawDocXattrsOthers)
+}
+
+func (t *RawDocXattrs) UnmarshalJSON(p []byte) error {
+	if err := json.Unmarshal(p, &t.RawDocXattrsWellKnown); err != nil {
+		return err
+	}
+	return json.Unmarshal(p, &t.RawDocXattrsOthers)
```

Comment:
In this case, I think you probably want to delete the keys of `_vv`, `_sync` etc, so these aren't listed twice.

Reply:
I'll add a TODO but probably just punt on this. It doesn't cause any issues for its current usage in tests and is only a problem if you go looking for the known keys inside the "Others" map, which is too awkward to accidentally do, vs. accessing the struct fields.

This will go away with the v2 implementation of JSON with the 'inline' feature - but this code is an awkward hack to get a set of known fields out, and then a map of everything else.

https://github.com/golang/go/issues/6213